src/QsCompiler/TextProcessor/QsFragmentParsing.fs

```diff
@@ -97,19 +97,21 @@ let private symbolBinding connector connectorErr expectedRhs = // used for mutab
 /// Raises a MissingInitializerExpression error or an InvalidInitializerError for missing or invalid initializers.
 /// Uses optTupleBrackets to raise the corresponding missing bracket errors if the entire assignment is not within tuple brackets.
 let private allocationScope = 
-    let combineRangeAndBuild (r1, (kind, r2)) = 
-        QsPositionInfo.WithCombinedRange (r1 |> QsPositionInfo.Range, r2) kind InvalidInitializer |> QsInitializer.New // *needs* to be invalid if the combined range is Null!
-    let qRegisterAlloc = 
-        qsQubit.parse .>>. (arrayBrackets (expectedExpr eof) 
-        |>> fun (ex, r) -> QubitRegisterAllocation ex, QsPositionInfo.Range r)
+    let combineRangeAndBuild (r1, (kind, r2)) =
+        match r2 with
+        | Value r2 -> { Initializer = kind; Range = Range.Span r1 r2 |> Value }
+        | Null -> { Initializer = InvalidInitializer; Range = Null }
```

Comment:
Same here, the comment should be preserved.

Reply:
Resolved in #579.